client/beefy/src/worker.rs

```diff
@@ -479,7 +492,14 @@ where
 			)?,
 			RoundAction::Enqueue => {
 				debug!(target: "beefy", "🥩 Buffer vote for round: {:?}.", block_num);
-				self.pending_votes.entry(block_num).or_default().push(vote)
+				if self.pending_votes.len() < MAX_BUFFERED_VOTE_ROUNDS {
+					let votes_vec = self.pending_votes.entry(block_num).or_default();
+					if votes_vec.try_push(vote).is_err() {
+						warn!(target: "beefy", "🥩 Buffer vote dropped for round: {:?}", block_num)
```

Comment:
Just random thought: value in `pending_votes` map may store duplicated votes, right? So in theory malicious validator may try to send `MAX_BUFFERED_VOTES_PER_ROUND` duplicate votes to all other validators, blocking round from concluding. If that's true, then it is something that must be tracked during equivocation impl.

Reply:
yes, that's correct